modules/analysis/src/endpoints/tests/mod.rs

```diff
@@ -335,6 +333,52 @@ async fn quarkus_component_by_purl(ctx: &TrustifyContext) -> Result<(), anyhow::
     Ok(())
 }
 
+/// find a component by query
+#[test_context(TrustifyContext)]
+#[test(actix_web::test)]
+async fn find_component_by_query(ctx: &TrustifyContext) -> Result<(), anyhow::Error> {
+    ctx.ingest_documents(["spdx/quarkus-bom-3.2.11.Final-redhat-00001.json"])
+        .await?;
+    const PURL: &str = "pkg:maven/com.redhat.quarkus.platform/quarkus-bom@3.2.11.Final-redhat-00001?repository_url=https://maven.repository.redhat.com/ga/&type=pom";
+
+    async fn query<T: AsRef<str>>(app: &impl CallService, query: T) {
+        let uri = format!("/api/v2/analysis/component?q={}", encode(query.as_ref()));
+        let request: Request = TestRequest::get().uri(&uri).to_request();
+        let response: Value = app.call_and_read_body_json(request).await;
+        tracing::debug!(test = "", "{response:#?}");
+        assert!(response.contains_subset(json!({
+            "items": [{
+                "purl": [ PURL ],
+                "cpe": ["cpe:/a:redhat:quarkus:3.2:*:el8:*"]
+            }]
+        })));
+    }
+
+    let app = caller(ctx).await?;
+    query(&app, "purl~quarkus-bom").await;
+    query(&app, "cpe=cpe:/a:redhat:quarkus:3.2:*:el8:*").await;
+    query(&app, "cpe~cpe:/a:redhat:quarkus:3.2").await;
+
+    // Testing for qualified purls is tricky, because the order of the
+    // qualifiers isn't predictable. One workaround is to "and" the
+    // qualifiers in the query using the LIKE operator,
+    // e.g. q=purl~BASE&purl~Q1&purl~Q2
+
+    // This won't work, due to the presence of qualifiers
+    // query(&app, format!("purl={}", PURL.replace("&", r"\&"))).await;
+
+    // This will work, but I was unable to include the
+    // `repository_url` qualifier in the query due to unexpected
+    // urlencoding, either in the database or the get_purl function.
+    query(
+        &app,
+        "purl~pkg:maven/com.redhat.quarkus.platform/quarkus-bom@3.2.11.Final-redhat-00001&purl~type=pom",
+    )
+    .await;
```

Comment:
@JimFuller-RedHat we need to decide whether this is acceptable. And if we think it's too much to expect users to remember the `~` (ILIKE) operator, we can easily make `=` behave like `~`. I'm thinking users may like that, but :shrug: